homeassistant/components/sonos/media_player.py

```diff
@@ -317,32 +317,32 @@ def media(self) -> SonosMedia:
         """Return the SonosMedia object from the coordinator speaker."""
         return self.coordinator.media
 
-    @property  # type: ignore[misc]
+    @property
     def media_content_id(self) -> str | None:
         """Content id of current playing media."""
         return self.media.uri
 
-    @property  # type: ignore[misc]
-    def media_duration(self) -> float | None:
+    @property
+    def media_duration(self) -> float | None:  # type: ignore[override]
         """Duration of current playing media in seconds."""
         return self.media.duration
 
-    @property  # type: ignore[misc]
-    def media_position(self) -> float | None:
+    @property
+    def media_position(self) -> float | None:  # type: ignore[override]
         """Position of current playing media in seconds."""
         return self.media.position
```

Comment:
Both `media_duration` and `media_position` should only return `int | None`. I.e. for `media_duration`
```py
    @property
    def media_duration(self) -> int | None:
        return int(self.media.duration) if self.media.duration else None
```